presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -458,15 +460,24 @@ private OuterJoinPushDownResult processLimitedOuterJoin(Expression inheritedPred
                 }
             }
 
-            // See if we can push down any outer or join predicates to the inner side
```

Comment:
Actually tests still pass if `joinPredicate` in that changed line.
Review from someone with more understanding of what is happening here would be awesome :)